### ISSUE TITLE
Discussed offline. This is a limitation to how unstable_dev works right now, not related to this refactor.

### ISSUE THREAD

**Issue by scruffymongrel** [cloudflare/workers-sdk#6622]
Discussed offline. This is a limitation to how unstable_dev works right now, not related to this refactor.
The problem is that unstable_dev will only read the dev-registry once, rather than polling it for changes.
So we need to make sure that upstream workers are in-place and ready before starting downstream workers.
The `wrangler dev` command does do polling, so does not have this problem. So normal users would not need such pauses.

Going forward we should add polling or watching to the unstable_dev stuff so that these are no longer needed.

_Originally posted by @petebacondarwin in https://github.com/cloudflare/workers-sdk/pull/5214#discussion_r1645876763_

**Comment by scruffymongrel** [comment]
Apologies, opened in error!
